cime_config/machines/config_machines.xml

```diff
@@ -3162,6 +3162,117 @@
+        <executable>mpirun</executable>
+        <arguments>
+           <arg name="num_tasks">--tag-output -n {{ total_tasks }}</arg>
+           <arg name="tasks_per_node"> --map-by ppr:{{ tasks_per_numa }}:socket:PE=$ENV{OMP_NUM_THREADS} --bind-to hwthread</arg>
+        </arguments>
+    </mpirun>
+    <module_system type="module" allow_error="true">
+       <init_path lang="sh">/soft/packaging/lmod/lmod/init/sh</init_path>
+       <init_path lang="csh">/soft/packaging/lmod/lmod/init/csh</init_path>
+       <init_path lang="python">/soft/packaging/lmod/lmod/init/env_modules_python.py</init_path>
+       <cmd_path lang="sh">module</cmd_path>
+       <cmd_path lang="csh">module</cmd_path>
+       <cmd_path lang="python">/soft/packaging/lmod/lmod/libexec/lmod python</cmd_path>
+       <modules>
+          <command name="purge"></command>
+          <command name="use">/soft/modulefiles</command>
+          <command name="load">spack cmake</command>
+          <command name="use">/soft/restricted/CNDA/updates/modulefiles</command>
+        </modules>
+        <modules compiler="!gnu">
+          <command name="load">oneapi/eng-compiler/2022.12.30.002</command>
+          <!--command name="load">oneapi/eng-compiler/2022.10.15.006</command-->
+          <!--command name="load">mpich/50.2/icc-all-pmix-gpu</command-->
+          <command name="load">intel_compute_runtime/release/22.43.24595.30</command>
+        </modules>
+        <modules compiler="gnu">
+           <command name="unload">spack cmake</command>
+           <command name="load">gcc/10.3.0</command>
+        </modules>
+	<modules>
+	  <command name="load">cray-pals</command>
+          <command name="load">append-deps/default</command>
+        </modules>
+     </module_system>
+     <RUNDIR>$CIME_OUTPUT_ROOT/$CASE/run</RUNDIR>
+     <EXEROOT>$CIME_OUTPUT_ROOT/$CASE/bld</EXEROOT>
+     <environment_variables>
+        <env name="NETCDF_PATH">/lus/gila/projects/CSC249ADSE15_CNDA/software/oneAPI.2022.12.30.003/netcdf</env>
+        <env name="PNETCDF_PATH">/lus/gila/projects/CSC249ADSE15_CNDA/software/oneAPI.2022.12.30.003/pnetcdf</env>
+     </environment_variables>
+     <environment_variables mpilib="mpich" DEBUG="TRUE">
+        <env name="HYDRA_TOPO_DEBUG">1</env>
+     </environment_variables>
+     <environment_variables mpilib="impi">
+        <env name="I_MPI_DEBUG">10</env>
+        <env name="I_MPI_PIN_DOMAIN">omp</env>
+        <env name="I_MPI_PIN_ORDER">spread</env>
+        <env name="I_MPI_PIN_CELL">unit</env>
+     </environment_variables>
+     <environment_variables compiler="oneapi-ifxgpu">
+        <env name="ONEAPI_DEVICE_SELECTOR">level_zero:gpu</env>
+        <env name="ONEAPI_MPICH_GPU">NO_GPU</env>
+        <env name="MPIR_CVAR_ENABLE_GPU">0</env>
+        <env name="romio_cb_read">disable</env>
+        <env name="romio_cb_write">disable</env>
+        <env name="SYCL_CACHE_PERSISTENT">1</env>
+        <env name="GATOR_INITIAL_MB">4000MB</env>
```

Comment:
@mrnorman Do you want us to set GATOR_INITIAL_MB in E3SM config or handle based on machine in YAKL?

Reply:
> 
It should be fine AoT in debug mode too, as the JIT and AoT build time are comparable at the moment, but the runtime has large differences

Reply:
That's up to you. YAKL default to consuming 1/8 of the total device memory. If you need to override that for any reason, you have two options: (1) `GATOR_INITIAL_MB`; or (2) passing a `yakl::InitConfig` object to `yakl::init()` specifying the runtime-determined pool sizes.

Whether or where to set `GATOR_INITIAL_MB`, I'm not sure. YAKL will not be doing anything machine-specific (i.e., no lookups for "frontier" or "summit" or anything in the `dnsdomainname` command. By default, YAKL will just look for the total amount device memory (or host memory for CPU targets) and allocate one eight of it for the initial pool and each additional pool.

Reply:
@xyuan Aren't you setting GATOR_INITIAL_MB to less than 1/8th here?